core/variant/type_info.h

```diff
@@ -220,31 +226,47 @@ inline String enum_qualified_name_to_class_info_name(const String &p_qualified_n
 	// Contains namespace. We only want the class and enum names.
 	return parts[parts.size() - 2] + "." + parts[parts.size() - 1];
 }
+
+template <typename T>
+inline String demangle_enum_name() {
+#ifdef __GNUC__
+	int status;
+	char *buffer = abi::__cxa_demangle(typeid(T).name(), nullptr, nullptr, &status);
```

Comment:
Is there a guarantee that this works even if the binary is obfuscated? I guess obfuscation may introduce other problems too, but just a thought.

Reply:
No clue. I'll dig for alternative ways of grabbing typenames natively just in case